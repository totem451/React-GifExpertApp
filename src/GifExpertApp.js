import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    // const categorias = ['Breacking bad', 'Peaky blinders', 'Suits'];

const [categorias, setCategorias] = useState(['Rick and Morty'/*, 'Breacking bad', 'Peaky blinders', 'Suits'*/]);

    // const handleAdd = () => {

    //     setCategorias([...categorias, 'Nueva serie']); // Agrega al final
    //     // setCategorias(['Nueva serie', ...categorias]); // Agrega al principio
    //     // setCategorias(c => [...c, 'Nueva serie']); // Agrega al final

    // };

    return (

        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategorias = {setCategorias}/>
            <hr />

            {/* <button onClick = {handleAdd}>Agregar</button> */}


            <ol>
                {
                    categorias.map(categoria => (

                        // <li key = {categoria}>{categoria}</li>
                        <GifGrid 
                            categoria = {categoria}
                            key = {categoria}
                        />
                    ))
                }
            </ol>
            {/* <>
                {
                    categorias.map((categoria, i) => {
                        return <li>{categorias[i]}</li>
                    })
                }
            </> */}
        </>

    );
}