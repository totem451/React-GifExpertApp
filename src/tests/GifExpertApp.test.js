import React from 'react';

import '@testing-library/jest-dom';
import '@wojtekmaj/enzyme-adapter-react-17';

import { shallow } from 'enzyme';

import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
    
    test('debe mostrarse correctamente ', () => {
        
        const wrapper = shallow(<GifExpertApp />);

        expect( wrapper ).toMatchSnapshot();

    });

    test('debe mostrar una lista de categorias ', () => {
        
        const categorias = ['Rick and Morty', 'weed'];
        const wrapper = shallow(<GifExpertApp defaultCategorias = { categorias } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categorias.length );
        
    })
    
})
