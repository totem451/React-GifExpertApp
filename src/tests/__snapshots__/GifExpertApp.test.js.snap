// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="Rick and Morty"
      key="Rick and Morty"
    />
    <GifGrid
      categoria="weed"
      key="weed"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe mostrarse correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
